lib/relations/Relation.js

```diff
@@ -148,6 +155,36 @@ Relation.prototype = {
         return fromPort !== toPort;
     },
 
+    get canonical() {
+        if (!('_canonical' in this)) {
+            var canonical = false;
+
+            if (this.assetGraph && this.assetGraph.canonicalRoot) {
+                var canonicalRootObj = urlModule.parse(this.assetGraph.canonicalRoot, false, true);
+                var hrefObj = urlModule.parse(this.href, false, true);
+
+                canonical = canonicalRootObj.host === hrefObj.host
+                    && hrefObj.path.indexOf(canonicalRootObj.path) === 0
+                    && (canonicalRootObj.protocol === hrefObj.protocol || canonicalRootObj.protocol === null);
+            }
+
+            this._canonical = canonical;
+        }
+
+        return this._canonical;
+    },
+
+    set canonical(isCanonical) {
+        this._canonical = !!isCanonical;
+
+        var hrefTypeRoot = this._canonical ? this.assetGraph.canonicalRoot : '';
+        // We need to update the hrefType manually here because otherwise hrefType in refreshHref
+        // will be based on the old href value, not the new one
+        this._hrefType = Object.getOwnPropertyDescriptor(Relation.prototype, 'hrefType').get.call({ href: hrefTypeRoot });
```

Comment:
This shouldn't be necessary as `hrefType` only expresses the intention to use eg. a relative href if possible -- `refreshHref` will still issue an absolute href if the host names differ.

If the `refreshHref` below doesn't do the right thing without the above two lines, I would say that it has a bug.